### PR TITLE
Move asset synchronization to IRemoteAssetSynchronizationService

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: src/VisualStudio/Core/Test.Next/Services/ServiceHubServicesTests.cs [hunk 1]
```diff
@@ -458,11 +454,9 @@ private static (Project, Document) GetProjectAndDocument(Solution solution, stri
 
         private async Task UpdatePrimaryWorkspace(RemoteHostClient client, Solution solution)
         {
-            await client.RunRemoteAsync(
-                WellKnownServiceHubService.RemoteHost,
-                nameof(IRemoteHostService.SynchronizePrimaryWorkspaceAsync),
+            await client.TryInvokeAsync<IRemoteAssetSynchronizationService>(
                 solution,
-                new object[] { await solution.State.GetChecksumAsync(CancellationToken.None), _solutionVersion++ },
+                async (service, solutionInfo, cancellationToken) => await service.SynchronizePrimaryWorkspaceAsync(solutionInfo, await solution.State.GetChecksumAsync(cancellationToken), _solutionVersion++, cancellationToken),
```
Comment (tmat) [inline src/VisualStudio/Core/Test.Next/Services/ServiceHubServicesTests.cs:459]: >await solution.State.GetChecksumAsync(cancellationToken) [](start = 129, length = 56)

I'd move `await solution.State.GetChecksumAsync(cancellationToken)` outside the lamda to simplify the expression.

File: src/VisualStudio/Core/Test.Next/Services/LanguageServiceTests.cs [hunk 2]
```diff
@@ -127,11 +127,9 @@ private async Task<ImmutableArray<SymbolInformation>> GetVsSearchResultsAsync(Te
 
         private async Task UpdatePrimaryWorkspace(RemoteHostClient client, Solution solution)
         {
-            await client.RunRemoteAsync(
-                WellKnownServiceHubService.RemoteHost,
-                nameof(IRemoteHostService.SynchronizePrimaryWorkspaceAsync),
+            await client.TryInvokeAsync<IRemoteAssetSynchronizationService>(
                 solution,
-                new object[] { await solution.State.GetChecksumAsync(CancellationToken.None), _solutionVersion++ },
+                async (service, solutionInfo, cancellationToken) => await service.SynchronizePrimaryWorkspaceAsync(solutionInfo, await solution.State.GetChecksumAsync(cancellationToken), _solutionVersion++, cancellationToken),
```
Comment (tmat) [inline src/VisualStudio/Core/Test.Next/Services/LanguageServiceTests.cs:132]: >await solution.State.GetChecksumAsync(cancellationToken) [](start = 129, length = 56)

Here too.

**Review by tmat — APPROVED** [review]
:shipit:

File: src/VisualStudio/Core/Test.Next/Services/ServiceHubServicesTests.cs (hunk 1 above)
Comment (sharwell) [reply src/VisualStudio/Core/Test.Next/Services/ServiceHubServicesTests.cs:459]: ➡️ Done

File: src/VisualStudio/Core/Test.Next/Services/LanguageServiceTests.cs (hunk 2 above)
Comment (sharwell) [reply src/VisualStudio/Core/Test.Next/Services/LanguageServiceTests.cs:132]: ➡️ Done

**Comment by sharwell** [comment]
@jinujoseph for approval for 16.9 Preview 1

**Comment by sharwell** [comment]
@msftbot merge if @jinujoseph approves

**Pull request merged**